### PR TITLE
[SSO] Reset master password 

### PR DESCRIPTION
**Pull request by vincentsalucci**
## Code Changes 
- **account/change-password.component.html**: Added html for new `change-password` (sso) flow
- **account/change-password.component.html**: Added ts class for new `change-password` (sso) flow - uses new base class. Updated `submit` flow to use callbacks.
- **sso.component.ts**: Removed all unnecessary code and converted class to use base class from jslib.
- **two-factor.component.ts**: Set query param to base class tracking boolean to make sure success route is properly set. Passed on `resetMasterPassword` boolean to the next route.
- **app-routing.module.ts**: Added new route
- **messages.json**: Added new strings
- **settings/change-password.component.ts**: Converted class to use new base class from `jslib` and also adapted new `submit` flow callbacks.
- **jslib**: Update from 7fa5178 to 34632e5

### CODE REVIEW

File: src/app/accounts/two-factor.component.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import {
     ViewContainerRef,
 } from '@angular/core';
 
-import { Router } from '@angular/router';
+import { Router, ActivatedRoute } from '@angular/router';
```
Comment (kspearrin) [inline src/app/accounts/two-factor.component.ts:8]: multiline

File: src/app/settings/change-password.component.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import {
     Component,
-    OnInit,
 } from '@angular/core';
 
-import { ToasterService } from 'angular2-toaster';
-import { Angulartics2 } from 'angulartics2';
+import {
+    Router,
+} from '@angular/router';
```
Comment (kspearrin) [inline src/app/settings/change-password.component.ts:7]: These could be single line

File: src/app/accounts/two-factor.component.ts (hunk 1 above)
Comment (vincentsalucci) [reply src/app/accounts/two-factor.component.ts]: Updated.

File: src/app/settings/change-password.component.ts (hunk 2 above)
Comment (vincentsalucci) [reply src/app/settings/change-password.component.ts]: Updated.

File: src/app/accounts/two-factor.component.ts [hunk 3]
```diff
@@ -66,7 +82,13 @@ export class TwoFactorComponent extends BaseTwoFactorComponent {
                 this.router.navigate([loginRedirect.route], { queryParams: loginRedirect.qParams });
                 await this.stateService.remove('loginRedirect');
             } else {
-                this.router.navigate([this.successRoute]);
+                this.router.navigate([this.successRoute],
```
[the remark this answers is not in the document]
Comment (vincentsalucci) [reply src/app/accounts/two-factor.component.ts]: Updated.

**Pull request merged**